test/doc/traits/callable_specs.cpp

```diff
@@ -0,0 +1,55 @@
+#include <iostream>
+#include <type_traits>
+#include <eve/traits/overload.hpp>
+#include <eve/wide.hpp>
+
+namespace eve
+{
+  // Defines two RABERU flags
+  using namespace rbr::literals;
+  inline constexpr auto precise = "precise"_fl;
+  inline constexpr auto scale   = "scale"_fl;
+
+  // Defines a support specification
+  struct precision
+  {
+    auto process_option(auto const& base, rbr::concepts::exactly<precise> auto opt) const
+    {
+      return options{rbr::merge(options{opt}, base)};
+    }
+
+    auto process_option(auto const& base, rbr::concepts::exactly<scale> auto opt) const
+    {
+      return options{rbr::merge(options{opt}, base)};
+    }
+
+    EVE_FORCEINLINE static constexpr auto defaults() noexcept { return options{};  }
+  };
+
+  // Make this callable supports the precision options
+  struct func_t : basic_callable<func_t>, supports<func_t, precision>
+  {
+    auto call(int) -> double;
+    EVE_CALLABLE_OBJECT(func_t, func_);
+  } inline constexpr func;
+};
+
+namespace eve::detail
+{
+  template<typename D>
+  auto func_(EVE_REQUIRES(cpu_), eve::options<D> const& opt, int x)
```

Comment:
Seems like we want a concept for options, what do you think? This D looks out of place.
Up to you, I don't care very much

Reply:
the concept is in rbr. I can propagate it to eve so it doesnt look horrible